llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21048,6 +21048,11 @@ bool RISCVTargetLowering::isLegalInterleavedAccessType(
       return false;
 
     ContainerVT = getContainerForFixedLengthVector(VT.getSimpleVT());
+  } else {
```

Comment:
Doesn't this apply to fixed vector types too? Don't we eventually convert the fixed vectors to scalable vectors using the same intrinsics?

Reply:
I was surprised by that too, but seemingly not, see the `load_factor2_as` and `store_factor2_as` tests I added.